### ISSUE TITLE
Use LLVM instead of GCC backend

### ISSUE THREAD

**Issue by rilysh** [pritamzope/xlang#4]
It seems like xlang uses libgcc as backend on compilation. Instead of libgcc, how about llvm code generation and link with a different linker? There's also several issues I noticed, like it forces to use curly braces wherever possible and explicit types when using functions like `printf`.

**Comment by Dashbloxx** [comment]
Is LLVM's backend faster?

**Comment by rilysh** [comment]
@Dashbloxx It's been couple months since I created this issue. I'm not sure how I made this issue, but to my realization, it doesn't directly link with libgcc, instead invokes NASM to assemble the assembly code and uses GCC to produce the output binary. 

I suggested to use LLVM as "direct" integration instead of using NASM and GCC. But the project goal seems different there. Also, LLVM was made to create compiler frontends and specific language compilers, both backends are fast. There's very little performance differences, which also varies in different use cases, and doesn't show up (in real world usage) unless both of them has passed micro-benchmarking with different results.
